dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -425,23 +452,44 @@ public void updateIndex(Context context) {
      */
     @Override
     public void updateIndex(Context context, boolean force) {
-        try {
-            Iterator<Item> items = null;
-            for (items = itemService.findAllUnfiltered(context); items.hasNext(); ) {
-                Item item = items.next();
-                indexContent(context, item, force);
-                //To prevent memory issues, discard an object from the cache after processing
-                context.uncacheEntity(item);
-            }
+        updateIndex(context, force, Constants.ITEM);
+        updateIndex(context, force, Constants.COLLECTION);
+        updateIndex(context, force, Constants.COMMUNITY);
+    }
 
-            List<Collection> collections = collectionService.findAll(context);
-            for (Collection collection : collections) {
-                indexContent(context, collection, force);
-            }
 
-            List<Community> communities = communityService.findAll(context);
-            for (Community community : communities) {
-                indexContent(context, community, force);
+    public void updateIndex(Context context, List<UUID> ids, boolean force, int type) {
+        if (type != Constants.ITEM) {
+            throw new RuntimeException("Only ITEM is supported in this mode - type founded: " + type);
+        }
+        try {
+            startMultiThreadIndex(context, force, ids);
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void updateIndex(Context context, boolean force, int type) {
+        try {
+            switch (type) {
+                case Constants.ITEM:
+                    startMultiThreadIndex(context, force, null);
+                    break;
+                case Constants.COLLECTION:
+                    List<Collection> collections = collectionService.findAll(context);
+                    for (Collection collection : collections) {
+                        indexContent(context, collection, force);
+                    }
+                    break;
+                case Constants.COMMUNITY:
+                    List<Community> communities = communityService.findAll(context);
+                    for (Community community : communities) {
+                        indexContent(context, community, force);
+                    }
+                    break;
+                default:
+                    throw new RuntimeException("No type known: " + type);
```

Comment:
Would an IllegalArgumentException not make more sense here instead of a generic "RuntimeException" as this is what it really is. An illegal Argument was passed along.

Reply:
done, https://github.com/4Science/DSpace/pull/65/commits/5e3164bd06e2eaefc0a047087b1ab7cebdb1ee4d